modules/GPTQ_loader.py

```diff
@@ -94,33 +104,52 @@ def load_quantized(model_name):
         print("Unknown pre-quantized model type specified. Only 'llama', 'opt' and 'gptj' are supported")
         exit()
 
-    # Now we are going to try to locate the quantized model file.
+    # Now we are going to try to locate the quantized model file. I think it's cleaner and supports the new name containing groupsize
     path_to_model = Path(f'{shared.args.model_dir}/{model_name}')
-    found_pts = list(path_to_model.glob("*.pt"))
-    found_safetensors = list(path_to_model.glob("*.safetensors"))
     pt_path = None
-
-    if len(found_pts) > 0:
-        pt_path = found_pts[-1]
-    elif len(found_safetensors) > 0:
-        pt_path = found_safetensors[-1]
-    else:
-        if path_to_model.name.lower().startswith('llama-7b'):
-            pt_model = f'llama-7b-{shared.args.wbits}bit'
-        elif path_to_model.name.lower().startswith('llama-13b'):
-            pt_model = f'llama-13b-{shared.args.wbits}bit'
-        elif path_to_model.name.lower().startswith('llama-30b'):
-            pt_model = f'llama-30b-{shared.args.wbits}bit'
-        elif path_to_model.name.lower().startswith('llama-65b'):
-            pt_model = f'llama-65b-{shared.args.wbits}bit'
+    priority_name_list = [
+        Path(f'{shared.args.model_dir}/{model_name}/{shared.args.wbits}bit-{shared.args.groupsize}g.safetensors'),
+        Path(f'{shared.args.model_dir}/{model_name}/{shared.args.wbits}bit-{shared.args.groupsize}g.pt'),
+        Path(f'{shared.args.model_dir}/{model_name}/{shared.args.wbits}bit.safetensors'),
+        Path(f'{shared.args.model_dir}/{model_name}/{shared.args.wbits}bit.pt'),
+        Path(f'{shared.args.model_dir}/{model_name}-{shared.args.wbits}bit-{shared.args.groupsize}g.safetensors'),
+        Path(f'{shared.args.model_dir}/{model_name}-{shared.args.wbits}bit-{shared.args.groupsize}g.pt'),
+        Path(f'{shared.args.model_dir}/{model_name}-{shared.args.wbits}bit.safetensors'),
+        Path(f'{shared.args.model_dir}/{model_name}-{shared.args.wbits}bit.pt'),
```

Comment:
This doesn't work if the model doesn't use `groupsize` and `shared.args.groupsize` is unset. Also, this is not a format that I have seen people use

```
models/llama-30b/4bit-128g.safetensors
models/llama-30b/4bit-128g.pt
models/llama-30b/4bit.safetensors
```

It's usually something like `modelname-4bit-128g.pt` or `modelname-4bit.pt`

Reply:
Not OP, but this format is used by the 0cc4m Kobold AI 4-bit branch, https://github.com/0cc4m/KoboldAI/tree/latestgptq

Reply:
@oobabooga I think an unset `shared.args.groupsize` is `-1`, and it leads to names of `xxx-4bit--1g.pt`, which is obviously not found and safely skipped.
What would you recommend on the searching order of model names? Like this?
```
Path(f'{shared.args.model_dir}/{model_name}/{model_name}-{shared.args.wbits}bit-{shared.args.groupsize}g.safetensors'),
Path(f'{shared.args.model_dir}/{model_name}/{model_name}-{shared.args.wbits}bit-{shared.args.groupsize}g.pt'),
Path(f'{shared.args.model_dir}/{model_name}/{model_name}-{shared.args.wbits}bit.safetensors'),
Path(f'{shared.args.model_dir}/{model_name}/{model_name}-{shared.args.wbits}bit.pt'),
Path(f'{shared.args.model_dir}/{model_name}-{shared.args.wbits}bit-{shared.args.groupsize}g.safetensors'),
Path(f'{shared.args.model_dir}/{model_name}-{shared.args.wbits}bit-{shared.args.groupsize}g.pt'),
Path(f'{shared.args.model_dir}/{model_name}-{shared.args.wbits}bit.safetensors'),
Path(f'{shared.args.model_dir}/{model_name}-{shared.args.wbits}bit.pt'),
```

Reply:
This could be cut down to just
```python
priority_name_list = [
	Path(f'{shared.args.model_dir}/{model_name}{hyphen}{shared.args.wbits}bit{group}{ext}')
	for ext in ['.safetensors', '.pt']
	for group in [f'-{shared.args.groupsize}g', '']
	for hyphen in ['-', '/']
]
```
which produces a list like
```python
[
	'dir/llama-4bit-128g.safetensors',
	'dir/llama/4bit-128g.safetensors',
	'dir/llama-4bit.safetensors',
	'dir/llama/4bit.safetensors',
	'dir/llama-4bit-128g.pt',
	'dir/llama/4bit-128g.pt',
	'dir/llama-4bit.pt',
	'dir/llama/4bit.pt'
]
```